pkg/controller/resourcedistribution/resourcedistribution_controller.go

```diff
@@ -0,0 +1,391 @@
+			} else if !ok {
+				if err := r.deleteResource(namespace, resourceName, oldResource); err != nil && !errors.IsNotFound(err) {
+					errCh <- &UnexpectedError{
+						err:         err,
+						namespace:   namespace,
+						conditionID: DeleteConditionID,
+					}
+					klog.Errorf("error occurred when deleting resource in namespace %s from client, err： %v, name: %s", namespace, err, distributor.Name)
+					return
+				}
+				klog.V(3).Infof("ResourceDistribution(%s) deleted (%s/%s) in namespaces %s", distributor.Name, oldResource.GetObjectKind(), resourceName, namespace)
+			}
+			if ok { // here means resource is distributed successfully
+				succeeded.Inc()
+			}
+		}(namespaceName)
+	}
+	wait.Wait() // wait for all distribution done.
+
+	// 6. process all unexpected errors
+	numberOfErrs := len(errCh)
+	if numberOfErrs != 0 {
+		err = fmt.Errorf("some unexpected errors occurred when distributing resource, error messages were recorded in .status.conditions, name: %s", distributor.Name)
+	}
+	for i := 0; i < numberOfErrs; i++ {
+		unexpected := <-errCh
+		setCondition(&conditions[unexpected.conditionID], unexpected.err, unexpected.namespace)
+	}
+
+	// 7. set condition for uncreated namespaces errors
+	setCondition(
+		&conditions[NoUncreatedNSConditionID],
+		fmt.Errorf("some target namespaces don't exist, please ensure namespace is created before distribution"),
+		uncreatedNamespaces...)
+
+	// 8. record warning events if there are failing distributions
+	done := succeeded.Load()
+	if done != newStatus.Desired {
+		r.recorder.Event(distributor, corev1.EventTypeWarning, "SomeDistributionFailed",
+			fmt.Sprintf("ResourceDistribution %s completed, %d distributions succeeded, but %d distributions failed", distributor.Name, done, newStatus.Desired-done))
+	}
+
+	return makeNewStatus(newStatus, distributor, conditions, done), err
+}
+
+// updateDistributorStatus update distributor status after reconcile
+func (r *ReconcileResourceDistribution) updateDistributorStatus(distributor *appsv1alpha1.ResourceDistribution) error {
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		updateErr := r.Client.Status().Update(context.TODO(), distributor)
+		if updateErr == nil {
+			return nil
+		}
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: distributor.Name}, distributor); err != nil {
+			klog.Errorf("getting error occurred when updating distributor %s status", distributor.Name)
+		}
+		return updateErr
+	}); err != nil {
+		return err
+	}
+
+	klog.V(3).Infof("ResourceDistribution(%s) update status success", distributor.Name)
+	return nil
+}
+
+// updateResource try to update resource
+func (r *ReconcileResourceDistribution) updateResource(distributor *appsv1alpha1.ResourceDistribution, namespace, name string, resource runtime.Object, hashCode string) error {
+	resourceObject := makeResourceObject(distributor, namespace, resource, hashCode)
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
It is an incorrect logic. What's the meaning you retry to get the object and directly update it?